harness/features/track.local.test.ts

```diff
@@ -0,0 +1,369 @@
+import { getConnectionStringForProxy, forEachSDK, describeIf, createClient, createUser, wait } from '../helpers'
+import { Capabilities, SDKCapabilities } from '../types'
+import { v4 as uuidv4 } from 'uuid'
+import { getServerScope } from '../mockServer'
+import { config } from '../mockData/index'
+import nock from 'nock'
+
+jest.setTimeout(1000000)
+
+const scope = getServerScope()
+
+describe('Track Tests - Local', () => {
+    const mockEvents = jest.fn()
+    const validUserId = 'user1'
+    forEachSDK((name) => {
+        let url: string
+        const capabilities: string[] = SDKCapabilities[name]
+        const clientId: string = uuidv4()
+        const mockServerUrl
+            = `http://${process.env.DOCKER_HOST_IP ?? 'host.docker.internal'}:${global.__MOCK_SERVER_PORT__}`
+
+        beforeAll(async () => {
+            url = getConnectionStringForProxy(name)
+            await createClient(url, clientId, 'dvc_server_test_token_parth2', {
+                baseURLOverride: `${mockServerUrl}/client/${clientId}`,
+                eventFlushIntervalMS: 1000, logLevel: 'debug'
+            })
+
+            scope
+                .get(`/client/${clientId}/config/v1/server/dvc_server_test_token_parth2.json`)
+                .reply(200, config)
+
+            scope.post(`/client/${clientId}`).reply((uri, body) => {
+                if (typeof body === 'object'
+                    && body.message.includes('onClientInitialized was invoked'))
+                    return [200]
+            })
+
+            // await wait(1000)
+        })
+
+        afterEach(() => {
+            jest.clearAllMocks()
+        })
+
+        // only for debugging
+        beforeEach(() => {
+            console.log('beforeEach', scope.activeMocks())
+        })
+
+        describeIf(capabilities.includes(Capabilities.cloud))(name, () => {
+            describe('Failing tests', () => {
+                it('should not send an event if the event type not set', async () => {
+
+                    const response = await createUser(url, { user_id: validUserId })
+                    await response.json()
+                    const userId = response.headers.get('location')
+
+                    const trackResponse = await callTrack(clientId, url, userId, {})
+
+                    scope.post(`/client/${clientId}/v1/events/batch`).reply((uri, body) => {
```

Comment:
you could also try not setting up a scope at all here and using the `no match` event in Nock to assert that no unexpected requests were made:
https://github.com/nock/nock#global-no-match-event

this would let you use the `scope.isDone` check as well since you wouldn't have set up an expected request